Assets/MixedRealityToolkit.Providers/UnityARCameraSettings/UnityARCameraSettings.cs

```diff
@@ -0,0 +1,257 @@
+            arCameraBackground = cameraObject.EnsureComponent<ARCameraBackground>();
+
+            trackedPoseDriver = cameraObject.EnsureComponent<TrackedPoseDriver>();
+
+            TrackedPoseDriver.TrackedPose poseSource;
+            TrackedPoseDriver.TrackingType trackingType;
+            TrackedPoseDriver.UpdateType updateType;
+
+            if (SettingsProfile != null)
+            {
+                // Read settings to be applied to the camera.
+                poseSource = SettingsProfile.PoseSource;
+                trackingType = SettingsProfile.TrackingType;
+                updateType = SettingsProfile.UpdateType;
+            }
+            else
+            {
+                Debug.LogWarning("A profile was not specified for the XR Camera Settings provider.\nApplying Microsoft Mixed Reality Toolkit default options.");
+                // Use default settings.
+                poseSource = TrackedPoseDriver.TrackedPose.ColorCamera;
+                trackingType = TrackedPoseDriver.TrackingType.RotationAndPosition;
+                updateType = TrackedPoseDriver.UpdateType.UpdateAndBeforeRender;
+            }
+
+            trackedPoseDriver.SetPoseSource(
+                TrackedPoseDriver.DeviceType.GenericXRDevice,
+                poseSource);
+            trackedPoseDriver.trackingType = trackingType;
+            trackedPoseDriver.updateType = updateType;
+            trackedPoseDriver.UseRelativeTransform = false;
+
+            isInitialized = true;
+        }
+
+        /// <summary>
+        /// Uninitialize and clean up AR Foundation components.
+        /// </summary>
+        private void UninitializeARFoundation()
+        {
+            if (!isInitialized) { return; }
+
+            if (!preExistingArSessionOriginObject &&
+                (arSessionOriginObject != null))
+            {
+                if (Application.isEditor && !Application.isPlaying)
+                {
+                    Object.DestroyImmediate(trackedPoseDriver);
+                    Object.DestroyImmediate(arCameraBackground);
+                    Object.DestroyImmediate(arCameraManager);
+                    Object.DestroyImmediate(arSessionOrigin);
+                }
+                else
+                {
+                    Object.Destroy(trackedPoseDriver);
+                    Object.Destroy(arCameraBackground);
+                    Object.Destroy(arCameraManager);
+                    Object.Destroy(arSessionOrigin);
+                }
+
+                trackedPoseDriver = null;
+                arCameraBackground = null;
+                arCameraManager = null;
+                arSessionOrigin = null;
+            }
+
+            if (!preExistingArSessionObject &&
+                (arSessionObject != null))
+            {
+                if (Application.isEditor && !Application.isPlaying)
+                {
+                    Object.DestroyImmediate(arInputManager);
```

Comment:
use GameObjectExtensions?

Reply:
these are components, not gameobjects. for now, let's leave as-is